### PR TITLE
[OAuth] Pass scope parameter also on token refresh

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/libsync/creds/oauth.cpp [hunk 1]
```diff
@@ -25,6 +25,11 @@
 #include "creds/httpcredentials.h"
 #include <QRandomGenerator>
 
+namespace {
+// TODO: make theming parameter
+QString SCOPE = QStringLiteral("openid offline_access email");
```
Comment (ogoffart) [inline src/libsync/creds/oauth.cpp]: One should avoid non-trivial static global object.
Use Q_GLOBAL_STATIC,  or return from a function, or just plain char*

File: src/libsync/creds/oauth.cpp [hunk 2]
```diff
@@ -101,7 +106,6 @@ void OAuth::startAuthentication()
                     { QStringLiteral("code"), args.queryItemValue(QStringLiteral("code")) },
                     { QStringLiteral("redirect_uri"), QStringLiteral("http://localhost:%1").arg(_server.serverPort()) },
                     { QStringLiteral("code_verifier"), _pkceCodeVerifier },
-                    { QStringLiteral("scope"), args.queryItemValue(QStringLiteral("scope")) },
```
Comment (ogoffart) [inline src/libsync/creds/oauth.cpp]: Wait, was that not the correct thing to do?

Comment (TheOneRing) [reply src/libsync/creds/oauth.cpp]: I have no idea and it looks like we would need the full set anyhow....
@DeepDiver1975 ?

Comment (ogoffart) [reply src/libsync/creds/oauth.cpp]: But then we would need to use `SCOPE()` here.
If you don't use it here, there is also no reason to use in the other call.

Comment (TheOneRing) [reply src/libsync/creds/oauth.cpp]: It's now set in postTokenRefresh

**Pull request merged**